src/lender/token/restricted.sol

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2020 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.15 <0.6.0;
+
+import "tinlake-erc20/erc20.sol";
+
+contract MemberlistLike {
+    function member(address) public;
+}
+
+contract RestrictedToken is ERC20 {
+
+    MemberlistLike public memberlist; 
+    modifier checkMember(address usr) { memberlist.member(usr); _; }
+    
+    function hasMember(address usr) public returns (bool) {
+        // will fail if user not member
+        memberlist.member(usr);
+    }
+
+    constructor(string memory symbol_, string memory name_) ERC20(symbol, name) public {
+        wards[msg.sender] = 1;
+    }
+
+    function depend(bytes32 contractName, address addr) public auth {
+        if (contractName == "memberlist") { memberlist = MemberlistLike(addr); }
+        else revert();
+    }
+
+    function transferFrom(address from, address to, uint wad) checkMember(to) public returns (bool) {
```

Comment:
We need to add a check for `from` as well.

Let's say I am in the list receive tokens. Later I get removed.  I could still send the tokens to another address afterwards.

Reply:
From the spec: `The token must only allow transfers to and from members in the list.`